pkg/crc/segment/segment_test.go

```diff
@@ -0,0 +1,55 @@
+package segment
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ConsentTelemetry = "consent-telemetry"
+)
+
+func newTestConfig(configFile string, value bool, envPrefix string) (*crcConfig.Config, error) {
+	storage, err := crcConfig.NewViperStorage(configFile, envPrefix)
+	if err != nil {
+		return nil, err
+	}
+	config := crcConfig.New(storage)
+	config.AddSetting(ConsentTelemetry, value, crcConfig.ValidateBool, crcConfig.SuccessfullyApplied)
+	return config, nil
+}
+
+func TestClientUploadWithoutConcent(t *testing.T) {
+	dir, err := ioutil.TempDir("", "cfg")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+	configFile := filepath.Join(dir, "crc.json")
+
+	config, err := newTestConfig(configFile, false, "CRC")
+	require.NoError(t, err)
+
+	c, err := NewClient(config)
```

Comment:
This client will use the real uuid and the real file. You need to use a temporary file. A test can verify the loading of the uuid or if empty, the generation of the uuid.